keps/sig-node/1867-disable-accelerator-usage-metrics/README.md

```diff
@@ -0,0 +1,179 @@
+- [X] (R) Design details are appropriately documented
+- [X] (R) Test plan is in place, giving consideration to SIG Architecture and SIG Testing input
+- [X] (R) Graduation criteria is in place
+- [X] (R) Production readiness review completed
+- [ ] Production readiness review approved
+- [ ] "Implementation History" section is up-to-date for milestone
+- [ ] User-facing documentation has been created in [kubernetes/website], for publication to [kubernetes.io]
+- [ ] Supporting documentation e.g., additional design documents, links to mailing list discussions/SIG meetings, relevant PRs/issues, release notes
+
+[kubernetes.io]: https://kubernetes.io/
+[kubernetes/enhancements]: https://git.k8s.io/enhancements
+[kubernetes/kubernetes]: https://git.k8s.io/kubernetes
+[kubernetes/website]: https://git.k8s.io/website
+
+## Summary
+
+This KEP outlines the process to deprecate the Accelerator Metrics collected by Kubelet.
+
+Accelerator metrics are no longer expected to collected by Kubelet, but by external monitoring agents using the PodResources API.
+The purpose of creating that API was to provide an out of tree mechanism for all vendors to provide device specific metrics.
+This allows them to provide these metrics without requiring them to make changes to Kubernetes.
+
+Now that this API is beta and soon to be G.A, this KEP outlines the process to deprecate metrics that were added before sig-node conveged on the PodResources API.
+
+## Motivation
+
+### Goals
+
+Deprecate and remove the AcceleratorUsage metrics that kubelet currently advertises.
+
+### Non-Goals
+
+Deprecation and removal of the summary API is certainly a goal of sig-node, and the step of removing the AcceleratorUsage metric is a step in that direction (agreed upon goal with incremental steps towards it) but not the goal of this KEP.
+
+## Proposal
+
+### Risks and Mitigations
+
+The main risk that we face is breaking existing consumers of the AcceleratorUsage metrics.
+The way this risk is mitigated is through the use of Feature Flags, which allows users to re-enable this metric.
+We will also be pointing in documentation users towards the newer and richer method of collecting metrics.
+
+Note that we don't know who are the consumers of that metric but we suspect that this will impact a small subset of users as these metrics on NVIDIA GPU utilization are often unreliable and very coarse.
+
+## Design Details
+
+Add a feature flag and pass the disable option to cadvisor.
+
+### Test Plan
+
+* E2E tests that checks when the feature flag is enabled if the metrics are present or not.
+
+### Graduation Criteria
+
+#### Alpha Graduation
+
+* Feature Flag is present.
+* E2E tests are implemented.
+* Documentation has been published on how to transition to the new metrics.
+* Release Notes have been created and promote immediate usage of that flag.
+
+#### Alpha -> Beta Graduation
+
+* Sufficient heads up has been given (1 year) to users.
```

Comment:
IMO, we can move to Beta after a single release, but should give a long-ish period before moving to GA.  Most users won't notice this until it is enabled by default, and thus if we want to give users time to adapt and migrate to the daemonset, it would be between Beta and GA, where they can explicitly opt-out of the deprecation.

Reply:
I'm in favor of doing that! Is that something others would be in favor of :) ?

Reply:
@derekwaynecarr @dchen1107

Reply:
ping @derekwaynecarr @dchen1107

Thanks !

Reply:
+1